src/instantsearch/instantsearch.ts

```diff
@@ -261,19 +254,11 @@ export class NgAisInstantSearch implements AfterViewInit, OnInit, OnDestroy {
       if (typeof config.routing !== 'undefined') delete config.routing;
     }
 
-    if (!config.searchClient && !config.createAlgoliaClient) {
-      const client = algoliasearch(config.appId, config.apiKey);
-      config.searchClient = client;
-      config.appId = undefined;
-      config.apiKey = undefined;
-    }
-
-    // custom algolia client agent
     if (typeof config.searchClient.addAlgoliaAgent === 'function') {
```

Comment:
we still need to check searchClient is defined before accessing addAlgoliaAgent

Reply:
The `searchClient` is required on the `config` object. It does not throw an error if it's not provided?